pkg/plugin/vector/embed/embedder.go

```diff
@@ -2,32 +2,36 @@ package embed
 
 import (
 	"context"
-
-	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
 )
 
 type EmbedderType string
 
 const (
-	EmbedderOpenAI EmbedderType = "openai"
+	EmbedderOpenAI    EmbedderType = "openai"
+	EmbedderVectorAPI EmbedderType = "grafana-vector-api"
 )
 
-type Embedder interface {
-	Embed(ctx context.Context, model string, text string) ([]float32, error)
-}
+type EmbedderAuthType string
+
+const (
+	EmbedderAuthTypeBasicAuth EmbedderAuthType = "basic-auth"
```

Comment:
These types feel like generic types e.g. `AuthTypeBasicAuth ` instead of `EmbedderAuthTypeBasicAuth`
Maybe move to settings.go?